slsDetectorSoftware/src/Module.cpp

```diff
@@ -2725,40 +2627,25 @@ void Module::setPartialFramesPadding(bool padding) {
 void Module::startReceiver() {
     LOG(logDEBUG1) << "Starting Receiver";
     shm()->stoppedFlag = false;
-    if (shm()->useReceiverFlag) {
-        sendToReceiver(F_START_RECEIVER, nullptr, nullptr);
-    }
+    sendToReceiver(F_START_RECEIVER, nullptr, nullptr);
 }
 
 void Module::stopReceiver() {
     LOG(logDEBUG1) << "Stopping Receiver";
-    if (shm()->useReceiverFlag) {
-        int arg = static_cast<int>(shm()->stoppedFlag);
-        sendToReceiver(F_STOP_RECEIVER, arg, nullptr);
-    }
+    int arg = static_cast<int>(shm()->stoppedFlag);
+    sendToReceiver(F_STOP_RECEIVER, arg, nullptr);
 }
 
 slsDetectorDefs::runStatus Module::getReceiverStatus() const {
-    runStatus retval = ERROR;
-    LOG(logDEBUG1) << "Getting Receiver Status";
-    if (shm()->useReceiverFlag) {
-        sendToReceiver(F_GET_RECEIVER_STATUS, nullptr, retval);
-        LOG(logDEBUG1) << "Receiver Status: " << ToString(retval);
-    }
-    return retval;
+    return sendToReceiver<runStatus>(F_GET_RECEIVER_STATUS);
 }
 
 int64_t Module::getFramesCaughtByReceiver() const {
-    int64_t retval = -1;
-    LOG(logDEBUG1) << "Getting Frames Caught by Receiver";
-    if (shm()->useReceiverFlag) {
-        sendToReceiver(F_GET_RECEIVER_FRAMES_CAUGHT, nullptr, retval);
-        LOG(logDEBUG1) << "Frames Caught by Receiver: " << retval;
-    }
-    return retval;
+    return sendToReceiver<int64_t>(F_GET_RECEIVER_FRAMES_CAUGHT);
 }
 
 std::vector<uint64_t> Module::getNumMissingPackets() const {
+    //TODO!(Erik) Refactor
     LOG(logDEBUG1) << "Getting num missing packets";
     if (shm()->useReceiverFlag) {
```

Comment:
You can also remove the check for useReceiverflag like you did for the other 'onlyreceiver' functions..

Reply:
The check is in, sendToReceiver and here we use the socket directly. Is there a reason we cant use sendToReceiver here?

Reply:
I guess knowing detector type and number of interfaces should help to know how many ports. Then we could stick to sendToReceiver (also in clientInterface). If you like, I can fix that after you push.